sstables/compaction_descriptor.hh

```diff
@@ -127,7 +127,7 @@ struct compaction_descriptor {
     std::vector<sstables::shared_sstable> sstables;
     // This is a snapshot of the table's sstable set, used only for the purpose of expiring tombstones.
     // If this sstable set cannot be provided, expiration will be disabled to prevent data from being resurrected.
-    std::optional<sstables::sstable_set> all_sstables_snapshot;
+    lw_shared_ptr<sstables::sstable_set> all_sstables_snapshot;
```

Comment:
Its confusing that the name says snapshot but the type is `sstable_set` even though there is an `sstable_set_snapshot` type now.

Reply:
It makes me doubt the correctness of this change.

Reply:
In the previous version the "snapshot" in this name actually meant a full copy, so it wouldn't be a snapshot as in sstable_set_snapshot right now, but I see that it may be very confusing now, so I'll adjust this name in the next version.